cctbx/french_wilson.py

```diff
@@ -336,10 +338,10 @@ def french_wilson_scale(
     raise Sorry(("The input intensities have uniform values (%g); this is probably "+
       "a bug in one of the data processing and/or conversion programs.") %
       miller_array.data()[0])
-  if params == None:
-    params = master_phil.extract()
-  if (params.max_bins is None) : # XXX reset in case of user error
-    params.max_bins = 60
+  if params and params.max_bins:
+    max_bins = params.max_bins
+  if params and params.min_bin_size:
+    min_bin_size = params.min_bin_size
```

Comment:
Maybe avoid checking `params` twice.
```suggestion
if params is not None:
  if params.max_bins is not None:
    max_bins = params.max_bins
  if params.min_bin_size is not None:
    min_bin_size = params.min_bin_size
```
Also, please add a comment that the default PHIL parameters are set in `master_phil` and that any changes to those default values should be reflected in the function definition and vice versa.

Reply:
I've left this as it is, as personally I find the version with only two `if` statements and one level of nesting more readable.